### DISCUSSION TITLE
Session Application Name

### DISCUSSION THREAD

**Discussion by dubsjw** [m1dugh/native-sound-mixer#19] [Q&A] [answered]
Is there any way to retrieve the application name for the session as it is displayed in the Windows Volume Mixer?  I'm looking for a solution/idea to always have a formatted name for the application that correlates to the session.  I'm using Electron/Typescript.  Any ideas are more than welcome.   Thanks.

**Comment by m1dugh** [comment, marked as answer]
Since you have the application path, you could parse it to retrieve the name of the app as displayed in Windows sound mixer app.

Example:
```
const reg = /\\[\w\.]+$/g

const path = "C:\\Myapp.test\\myapp.xyz.exe" // any path retrieved from native-sound-mixer

const name = path.match(reg).pop().substring(1)

console.log(name) // myapp.xyz.exe
const subname = name.split(".")[0]
console.log(subname) // myapp
```

Hope it was the answer you were looking for.
